### DISCUSSION TITLE
This ngx-scrollbar is not working for angular 16 + versions..

### DISCUSSION THREAD

**Discussion by shinkark** [community/community#120705] [Programming Help]
### Select Topic Area

Question

### Body

While installing this plugin in angular 16 that time got error
Unhandled Promise rejection: PerfectScrollbarModule does not have a module def (ɵmod property) ; Zone: <root> ; Task: Promise.then ; Value: Error: PerfectScrollbarModule does not have a module def (ɵmod property)

..any solution for this??

**Comment by david-wei-01001** [comment]
The error message suggests that there might be a compatibility issue with the version of `ngx-scrollbar` (or `PerfectScrollbarModule`, depending on the library you are actually using) you're trying to use in your Angular 16 project. Angular frequently updates its internal APIs and mechanics, which can sometimes lead to compatibility issues with libraries that were built for older versions of Angular.

Here are some steps you can take to resolve this issue:

1. **Check for Updates**: Look for the latest version of the library that is compatible with Angular 16. Library maintainers often update their packages to keep up with the latest Angular releases.

2. **Verify Installation**: Ensure that you've followed the installation instructions for the library correctly. This includes checking for any peer dependencies that might need to be installed.

3. **Look for Alternatives**: If the library hasn't been updated for Angular 16, consider looking for alternative libraries that provide similar functionality and are compatible with Angular 16.

4. **Check Open Issues**: Look at the library's repository on GitHub (or wherever the code is hosted) to check if this is a known issue. You might find that the maintainer is already working on a fix, or you might find a workaround suggested by another user.

5. **Downgrade Temporarily**: If you have the option and the library is essential for your project, you might consider temporarily downgrading your Angular version to the latest one that is known to work with the library, until an update is available.

6. **File an Issue**: If there isn't already an open issue for this problem, consider filing one in the library's repository. Provide as much detail as you can, including the error message, Angular version, and library version.

7. **Local Patch**: As a last resort, if you need a quick fix and you're comfortable doing so, you can try to patch the library locally. This would involve forking the library, making changes to the module definition to make it compatible with Angular 16, and then using your forked version in your project.

Please note that the exact solution may vary depending on the specifics of the library you are using and the nature of the incompatibility with Angular 16. If the library is `ngx-scrollbar` and you're getting an error related to `PerfectScrollbarModule`, it's also possible that you're facing an issue due to mismatched libraries or a missing step in the setup process. Make sure to review any available migration guides for the library when upgrading to Angular 16.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @shinkark !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Feed category is solely related to conversations around the GitHub product Feed. This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
